### PR TITLE
#167313415 Users should be able to see time it takes to read an article functionality

### PR DESCRIPTION
**Pull request by minega25**
#### What does this PR do?
Gives users the ability to see the time it takes to read an article
#### How should this be manually tested?
Endpoints to test
`GET /api/v1/articles/`
`GET /api/v1/articles/:slug`
#### Any background context you want to provide?
How it works:
- Read time is based on the average reading speed of an adult (roughly 265 WPM)
- We take the total word count of a post and translate it into minutes.
#### What are the relevant pivotal tracker stories?
[#167313415](https://www.pivotaltracker.com/story/show/167313415)
#### Screenshots (if appropriate)
Screen Shot 2019-08-20 at 16 39 26
Screen Shot 2019-08-20 at 16 40 24

### CODE REVIEW

**Review by salviosage — APPROVED** [review]
LGTM

File: src/controllers/articles.controller.js [hunk 1]
```diff
@@ -105,9 +116,13 @@ class Articles {
         message: 'That article does not exist!'
       });
     }
+    const article = _.pick(findArticle, ['slug', 'title', 'description', 'body', 'taglist', 'favorited', 'favoritedcount', 'flagged', 'images', 'views']);
+    const readTime = Helper.calculateReadTime(article.body);
```
Comment (nshutijonathan) [inline src/controllers/articles.controller.js:120]: Where are You calling .calculateReadTime from ? I don't see it in the helper middleware

File: test/test.articles.js [hunk 2]
```diff
@@ -43,4 +43,28 @@ describe('Articles', () => {
...
+      .get('/api/v1/articles')
+      .set('Content-Type', 'application/json')
+      .set('Authorization', usertoken)
+      .end((err, res) => {
+        expect(res.status).to.be.deep.equal(200);
+        expect(res.body).to.have.deep.property('message', 'List of all articles');
+      });
+    done();
+  });
+  it('view single article', (done) => {
```
Comment (Mnickii) [inline test/test.articles.js:58]: Don't think this test is relevant to your feature..

File: src/controllers/articles.controller.js (hunk 1 above)
Comment (minega25) [reply src/controllers/articles.controller.js:120]: The calculateReadTime method is in `./src/helpers/helper.js`

File: test/test.articles.js [hunk 3]
```diff
@@ -43,4 +43,28 @@ describe('Articles', () => {
...
+    chai
+      .request(server)
+      .get('/api/v1/articles/fakeslug')
+      .set('Content-Type', 'application/json')
+      .set('Authorization', usertoken)
+      .end((err, res) => {
+        expect(res.status).to.be.deep.equal(200);
+        expect(res.body).to.have.deep.property('message', 'Article successfully retrieved');
```
Comment (Mnickii) [inline test/test.articles.js:66]: Nor this

**Review by Mnickii — CHANGES_REQUESTED** [review]
Cannot see a test for your changes. Where is the test for calculate read time?

File: test/test.articles.js (hunk 2 above)
Comment (minega25) [reply test/test.articles.js:58]: I think the person who worked on the article CRUD did not create tests for getting article. So by adding my code in the controller, I was decreasing test coverage by 0.7%

**Comment by minega25** [comment]
> Cannot see a test for your changes. Where is the test for calculate read time?

The test for that function is in helper.test.js file. 
Git is rendering this file as a generated file, Click the `load diff` button. I think this is because am passing a variable that has a value that is really long, in order test count words functionality

File: test/test.articles.js (hunk 3 above)
Comment (minega25) [reply test/test.articles.js:66]: I think the person who worked on the article CRUD did not create tests for getting article. So by adding my code in the controller, I was decreasing test coverage by 0.7%

**Review by Mnickii — APPROVED** [review]
LGTM

**Review by nkalyesubula — APPROVED** [review]
Nice work Patrick!

**Review by habinezadalvan — APPROVED** [review]

**Pull request merged**